Gems/ScriptCanvas/Code/Include/ScriptCanvas/Grammar/ParsingUtilitiesScriptEventExtension.cpp

```diff
@@ -0,0 +1,244 @@
+#include <ScriptCanvas/Variable/VariableBus.h>
+#include <ScriptCanvas/Variable/VariableData.h>
+#include <ScriptEvents/ScriptEventsMethod.h>
+
+namespace ParsingUtilitiesScriptEventExtensionCpp
+{
+    using namespace ScriptEvents;
+    using namespace ScriptCanvas;
+    using namespace ScriptCanvas::Nodes::Core;
+
+    AZ::Outcome<ScriptEvents::Method, AZStd::string> TranslateToScriptEventMethod(const FunctionDefinitionNode& node)
+    {
+        ScriptEvents::Method method;
+        method.GetNameProperty().Set(node.GetDisplayName());
+
+        auto parameterSlots = node.GetSlotsByType(CombinedSlotType::DataOut);
+        for (auto& slot : parameterSlots)
+        {
+            auto& parameter = method.NewParameter();
+            parameter.GetNameProperty().Set(slot->GetName());
+            auto azType = slot->GetDataType().GetAZType();
+            parameter.GetTypeProperty().Set(azType);
+        }
+                        
+        auto methodValidateOutcome = method.Validate();
+        if (!methodValidateOutcome.IsSuccess())
+        {
+            return AZ::Failure(methodValidateOutcome.GetError());
+        }
+
+        auto slots = node.GetSlotsByType(CombinedSlotType::ExecutionOut);
+        if (slots.size() > 1)
+        {
+            return AZ::Failure(AZStd::string("Event nodes must have one or zero Execution Out Slots"));
+        }
+
+        if (!slots.empty())
+        {
+            auto resultNodes = node.GetConnectedNodes(*slots.front());
+            if (!resultNodes.empty())
+            {
+                if (resultNodes.size() > 1)
+                {
+                    return AZ::Failure(AZStd::string("Event nodes can only have one connected return value Node"));
+                }
+
+                auto returnValueNode = azrtti_cast<const Nodes::Core::FunctionDefinitionNode*>(resultNodes.front().first);
+                if (!returnValueNode || !returnValueNode->IsExecutionExit())
+                {
+                    return AZ::Failure(AZStd::string("Event nodes can only be connected to a FunctionDefinitionNode that defines a return value"));
+                }
+
+                auto returnValueSlots = returnValueNode->GetSlotsByType(CombinedSlotType::DataIn);
+                if (returnValueSlots.size() != 1)
+                {
+                    return AZ::Failure(AZStd::string("Event nodes can only be connected to a FunctionDefinitionNode that defines a single return value slot."));
+                }
+
+                for (auto& slot : returnValueSlots)
+                {
+                    const auto azTypeReturn = slot->GetDataType().GetAZType();
+                    method.GetReturnTypeProperty().Set(azTypeReturn);
+                }
+            }
+        }
+
+        return AZ::Success(method);
+    }
+}
+
+namespace ScriptCanvas::ScriptEventGrammar
+{
+    using namespace ScriptCanvas;
+    using namespace ParsingUtilitiesScriptEventExtensionCpp;
```

Comment:
Repeat: the `using namespace ScriptCanvas;` call needs to be moved to function scope or removed altogether